src/swarm/neo/util/Batch.d

```diff
@@ -0,0 +1,617 @@
+
+        // Compress into destination.
+        auto dst = compress_buf[size_t.sizeof .. $];
+        auto compressed_len = lzo.compress(*this.buffer, dst);
+
+        // Minimize dest length and return.
+        compress_buf.length = compressed_len + size_t.sizeof;
+        enableStomping(compress_buf);
+        return compress_buf;
+    }
+
+    /***************************************************************************
+
+        Removes all records from the batch.
+
+    ***************************************************************************/
+
+    public void clear ( )
+    {
+        this.buffer.length = 0;
+        enableStomping(*this.buffer);
+    }
+
+    /***************************************************************************
+
+        Calculates the size in the batch required by a single record field.
+
+        Params:
+            Field = type of field
+            field = field to calculate batched size of
+
+        Returns:
+            the number of bytes the specified field would consume in the batch
+
+    ***************************************************************************/
+
+    private size_t sizeOfField ( Field ) ( Field field )
+    {
+        static if ( isDynamicArrayType!(Field) )
+            return size_t.sizeof +
+                (field.length * ElementTypeOfArray!(Field).sizeof);
+        else
+            return Field.sizeof;
+    }
+
+    /***************************************************************************
+
+        Adds a single record field to the batch.
+
+        Params:
+            Field = type of field
+            field = field to add to the batch
+
+    ***************************************************************************/
+
+    private void addField ( Field ) ( Field field )
+    {
+        static if ( isDynamicArrayType!(Field) )
+        {
+            auto len = field.length;
+            this.addBytes((cast(void*)&len)[0 .. size_t.sizeof]);
+            this.addBytes(field[]);
+        }
+        else
+            this.addBytes((cast(void*)&field)[0 .. Field.sizeof]);
+    }
+
+    /***************************************************************************
+
+        Adds the specified bytes to the batch.
+
+        Params:
+            data = slice of data to add to the batch
+
+    ***************************************************************************/
+
+    private void addBytes ( in void[] data )
+    {
+        (*this.buffer) ~= data;
+    }
+}
+
+/*******************************************************************************
+
+    Record batch extractor.
+
+    Params:
+        Record = tuple of types of which a record consists
+
+*******************************************************************************/
+
+public scope class BatchReader ( Record ... )
+{
+    import ocean.core.Enforce;
+
+    static assert(recordFieldsSupported!(Record)());
+
+    /// Batch. Set in constructor.
+    private Const!(void)[] batch;
```

Comment:
This doesn't seem to be used anywhere.

Reply:
:+1: Updated. (Also added comment to ctor arg about lifetime of buffer.)